homework/homework5/treap.py

```diff
@@ -0,0 +1,73 @@
+from collections.abc import MutableMapping
+from typing import Optional, Iterator, Any
+
+from homework.homework5.node import Node
+
+
+class Treap(MutableMapping):
+    __size: int = 0
+    __head: Optional[Node] = None
+
+    def __str__(self) -> str:
+        return str(self.__head)
+
+    def __setitem__(self, key, value) -> None:
+        if key in self:
+            assert self.__head is not None
```

Comment:
I prefer to have some messages about errors

Reply:
I did this so there would be no problems with mypy. If the key is present in the treap, the `__head` is definitely not None.